virtual_rainforest/models/plants/canopy.py

```diff
@@ -20,7 +20,7 @@
 
 def generate_canopy_model(
     community: list[PlantCohort],
-) -> tuple[NDArray, NDArray, NDArray]:
+) -> tuple[NDArray, NDArray, NDArray, NDArray]:
```

Comment:
This return value is getting a bit cumbersome. Would it be worth creating a new dataclass object to house these values so they can be named? I know it's a bit more boilerplate, but I think it might make the interface cleaner.

Reply:
Yeah - I think that makes sense, particularly since this is going to get more cumbersome as more resources (fruit mass, etc) are added.

Reply:
Actually - this probably is the right way to go, but things are going to change radically as this model introduces some actual science and doesn't  just assert values, and this would make the internals of the functions more complex. So I'm going to park it for now.